### PR TITLE
Support gzip compression

### PR DESCRIPTION
**Pull request by myungsegyo**
Hello guys.
I added gzip compression support.
This is my first PR so please let me know if there are any mistakes.

### CODE REVIEW

**Review by WisdomPill — APPROVED** [review]
LGTM!

just please add a changeling file

**Comment by myungsegyo** [comment]
@WisdomPill 
Thank you for your review
I added a changeling file

**Comment by WisdomPill** [comment]
@myungsegyo could you rebase your PR please?

**Comment by myungsegyo** [comment]
@WisdomPill 
I rebased my branch but some of tests fail until now.
Is there something for me to do?

File: django_redis/compressors/gzip.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    def compress(self, value: bytes) -> bytes:
+        if len(value) > self.min_length:
+            return gzip.compress(value)
+        return value
+
+    def decompress(self, value: bytes) -> bytes:
+        try:
+            return gzip.decompress(value)
+        except gzip.BadGzipFile as e:
+            raise CompressorError(e)
```
Comment (WisdomPill) [inline django_redis/compressors/gzip.py:19]:
```suggestion
        except gzip.BadGzipFile as e:
            raise CompressorError from e
```

Comment (WisdomPill) [reply django_redis/compressors/gzip.py:19]: this will fix ruff's error

File: django_redis/compressors/gzip.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    def compress(self, value: bytes) -> bytes:
+        if len(value) > self.min_length:
+            return gzip.compress(value)
+        return value
+
+    def decompress(self, value: bytes) -> bytes:
+        try:
+            return gzip.decompress(value)
+        except gzip.BadGzipFile as e:
```
Comment (WisdomPill) [inline django_redis/compressors/gzip.py:18]: this is not available for python 3.6 and 3.7, read [here](https://docs.python.org/3.8/library/gzip.html#gzip.BadGzipFile)

**Comment by myungsegyo** [comment]
@WisdomPill 
I think this PR can be merged now!
Thank you for your kind review.

**Pull request merged**